[CoordinatedGraphics] Simplify drawing area resize implementation

carlosgcampos opened the pull request:
#### 44655c8e68f9087772f26f8935f5e1a1280f0c33
<pre>
[CoordinatedGraphics] Simplify surface resize
<a href="https://bugs.webkit.org/show_bug.cgi?id=285991">https://bugs.webkit.org/show_bug.cgi?id=285991</a>

Reviewed by NOBODY (OOPS!).

We currently have a method in the threaded compositor to set the new
size, which schedules an update without waiting for the layer flush. We
can pass the size and device scale factor as part of the composition
request, which always happens after the layer flush. Then the compositor
always tries to resize the surface to the requested size, and only when
it changed it calls glViewport with the new size.
This patch also handles the completion handler of
DrawingAreaProxyCoordinatedGraphics::deviceScaleFactorDidChange()
correctly making SetDeviceScaleFactor async like in apple ports and
waiting for the response to call the completion handler.

* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp:
(WebKit::DrawingAreaProxyCoordinatedGraphics::deviceScaleFactorDidChange):
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::AcceleratedSurface):
(WebKit::AcceleratedSurface::resize):
(WebKit::AcceleratedSurface::hostResize): Deleted.
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.h:
(WebKit::AcceleratedSurface::clientResize): Deleted.
(WebKit::AcceleratedSurface::size const): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::setDeviceScaleFactor):
(WebKit::DrawingAreaCoordinatedGraphics::deviceOrPageScaleFactorChanged): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::flushLayers):
(WebKit::LayerTreeHost::sizeDidChange):
(WebKit::LayerTreeHost::deviceOrPageScaleFactorChanged): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostTextureMapper.cpp:
(WebKit::LayerTreeHost::deviceOrPageScaleFactorChanged): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostTextureMapper.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::create):
(WebKit::ThreadedCompositor::ThreadedCompositor):
(WebKit::ThreadedCompositor::renderLayerTree):
(WebKit::ThreadedCompositor::requestComposition):
(WebKit::ThreadedCompositor::setViewportSize): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.h:
* Source/WebKit/WebProcess/WebPage/DrawingArea.h:
(WebKit::DrawingArea::updateBackingStoreState):
(WebKit::DrawingArea::setDeviceScaleFactor):
* Source/WebKit/WebProcess/WebPage/DrawingArea.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didScalePage):
(WebKit::WebPage::setDeviceScaleFactor):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::resize):
(WebKit::AcceleratedSurfaceDMABuf::clientResize): Deleted.
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.h:
* Source/WebKit/WebProcess/WebPage/libwpe/AcceleratedSurfaceLibWPE.cpp:
(WebKit::AcceleratedSurfaceLibWPE::AcceleratedSurfaceLibWPE):
(WebKit::AcceleratedSurfaceLibWPE::initialize):
(WebKit::AcceleratedSurfaceLibWPE::resize):
(WebKit::AcceleratedSurfaceLibWPE::clientResize): Deleted.
* Source/WebKit/WebProcess/WebPage/libwpe/AcceleratedSurfaceLibWPE.h:
* Source/WebKit/WebProcess/WebPage/wc/DrawingAreaWC.h:
</pre>

fujii commented on Source/WebKit/WebProcess/WebPage/DrawingArea.messages.in:
```diff
@@ -22,8 +22,9 @@
 messages -> DrawingArea NotRefCounted {
 #if USE(COORDINATED_GRAPHICS) || USE(TEXTURE_MAPPER)
-    UpdateBackingStoreState(uint64_t backingStoreStateID, bool respondImmediately, float deviceScaleFactor, WebCore::IntSize size, WebCore::IntSize scrollOffset)
+    UpdateGeometry(WebCore::IntSize size) -> ()
```
WinCairo EWS failed. WinCairo is already using `UpdateGeometry`. Could you rename either message?

carlosgcampos committed:
[CoordinatedGraphics] Simplify drawing area resize implementation
https://bugs.webkit.org/show_bug.cgi?id=255392

Reviewed by Žan Doberšek.

We currently have a backing store ID that we change every time the
drawing area is resized or new backing store is created and sent to the
web process waiting for a response. It can be simplified by using that
only for size changes and use an message with async reply to handle the
response.

* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp:
(WebKit::DrawingAreaProxyCoordinatedGraphics::paint): Just return early if we don't have a backing store to paint.
(WebKit::DrawingAreaProxyCoordinatedGraphics::sizeDidChange): Call sendUpdateGeometry() if not already sent.
(WebKit::DrawingAreaProxyCoordinatedGraphics::deviceScaleFactorDidChange): Send SetDeviceScaleFactor message.
(WebKit::DrawingAreaProxyCoordinatedGraphics::update): If waiting for geometry update and size doesn't match just ignore the update.
(WebKit::DrawingAreaProxyCoordinatedGraphics::enterAcceleratedCompositingMode): backingStoreStateID is no longer used.
(WebKit::DrawingAreaProxyCoordinatedGraphics::exitAcceleratedCompositingMode): Ditto.
(WebKit::DrawingAreaProxyCoordinatedGraphics::updateAcceleratedCompositingMode): Ditto.
(WebKit::DrawingAreaProxyCoordinatedGraphics::incorporateUpdate): Create a new backing store if the size changed.
(WebKit::DrawingAreaProxyCoordinatedGraphics::sendUpdateGeometry): Send UpdateGeometry message to the web process.
(WebKit::DrawingAreaProxyCoordinatedGraphics::didUpdateGeometry): If the size has changed while waiting for the geometry update call
sendUpdateGeometry() again.
(WebKit::DrawingAreaProxyCoordinatedGraphics::discardBackingStore):
(WebKit::DrawingAreaProxyCoordinatedGraphics::didUpdateBackingStoreState): Deleted.
(WebKit::DrawingAreaProxyCoordinatedGraphics::backingStoreStateDidChange): Deleted.
(WebKit::DrawingAreaProxyCoordinatedGraphics::sendUpdateBackingStoreState): Deleted.
(WebKit::DrawingAreaProxyCoordinatedGraphics::waitForAndDispatchDidUpdateBackingStoreState): Deleted.
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.h:
* Source/WebKit/UIProcess/DrawingAreaProxy.h:
(WebKit::DrawingAreaProxy::update):
(WebKit::DrawingAreaProxy::didUpdateBackingStoreStateTimeout): Deleted.
(WebKit::DrawingAreaProxy::didUpdateBackingStoreState): Deleted.
* Source/WebKit/UIProcess/DrawingAreaProxy.messages.in: Remove DidUpdateBackingStoreState message.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::forceRepaint): We don't need to force repaint again after size change,
since size change already forces a repaint.
(WebKit::DrawingAreaCoordinatedGraphics::setDeviceScaleFactor): Notify the WebPage.
(WebKit::DrawingAreaCoordinatedGraphics::setRootCompositingLayer):
(WebKit::DrawingAreaCoordinatedGraphics::layerHostDidFlushLayers):
(WebKit::DrawingAreaCoordinatedGraphics::updateGeometry): Set the new size and force a repaint.
(WebKit::DrawingAreaCoordinatedGraphics::enterAcceleratedCompositingMode):
(WebKit::DrawingAreaCoordinatedGraphics::sendEnterAcceleratedCompositingModeIfNeeded):
(WebKit::DrawingAreaCoordinatedGraphics::exitAcceleratedCompositingMode):
(WebKit::DrawingAreaCoordinatedGraphics::display):
(WebKit::DrawingAreaCoordinatedGraphics::updateBackingStoreState): Deleted.
(WebKit::DrawingAreaCoordinatedGraphics::sendDidUpdateBackingStoreState): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.h:
* Source/WebKit/WebProcess/WebPage/DrawingArea.h:
(WebKit::DrawingArea::setDeviceScaleFactor):
* Source/WebKit/WebProcess/WebPage/DrawingArea.messages.in:

Canonical link: https://commits.webkit.org/263118@main
